packages/flutter/lib/src/material/snack_bar.dart

```diff
@@ -245,10 +246,57 @@ class SnackBar extends StatelessWidget {
   /// The animation driving the entrance and exit of the snack bar.
   final Animation<double> animation;
 
+  /// Called the first time that the snackbar is visible within a [Scaffold].
+  final VoidCallback onVisible;
+
+  // API for Scaffold.addSnackBar():
+
+  /// Creates an animation controller useful for driving a snack bar's entrance and exit animation.
+  static AnimationController createAnimationController({ @required TickerProvider vsync }) {
+    return AnimationController(
+      duration: _snackBarTransitionDuration,
+      debugLabel: 'SnackBar',
+      vsync: vsync,
+    );
+  }
+
+  /// Creates a copy of this snack bar but with the animation replaced with the given animation.
+  ///
+  /// If the original snack bar lacks a key, the newly created snack bar will
+  /// use the given fallback key.
+  SnackBar withAnimation(Animation<double> newAnimation, { Key fallbackKey }) {
+    return SnackBar(
+      key: key ?? fallbackKey,
+      content: content,
+      backgroundColor: backgroundColor,
+      elevation: elevation,
+      shape: shape,
+      behavior: behavior,
+      action: action,
+      duration: duration,
+      animation: newAnimation,
+      onVisible: onVisible,
+    );
+  }
+
+  @override
+  State createState() => _SnackBarState();
+}
+
+
+class _SnackBarState extends State<SnackBar> {
+  @override
+  void initState() {
+    super.initState();
+    if (widget.onVisible != null) {
```

Comment:
It feels very odd doing this in the initState. I wonder if it would be better to tie this to the animation? When the animation is complete, the snackbar is visible?

/cc @HansMuller

Reply:
That should work, modulo handling the disable animation from accessibility

Reply:
I've added it as an animation status listener